packages/backend/src/shared/utils.ts

```diff
@@ -171,6 +172,64 @@ async function canApiSignatureBeProcessed(
   return true
 }
 
+/*
+  Verifies http signatures by first attempting to replicate it with a secret
+  associated with a tenant id in the headers, then with the configured admin secret.
+
+  If a tenant secret can replicate the signature, the request is tenanted to that particular tenant.
+  If the environment admin secret replicates the signature, then it is an operator request with elevated permissions.
+  If neither can replicate the signature then it is unauthorized.
+*/
+export async function verifyTenantOrOperatorApiSignature(
+  ctx: TenantedHttpSigContext,
+  config: IAppConfig
+): Promise<boolean> {
+  ctx.tenant = undefined
+  ctx.isOperator = false
+  const { headers } = ctx.request
+  const signature = headers['signature']
+  if (!signature) {
+    return false
+  }
+
+  const tenantService = await ctx.container.use('tenantService')
+  const tenantId = headers['tenantid']
+  const tenant = tenantId ? await tenantService.get(tenantId) : undefined
```

Comment:
should we return false if we can't find the tenant?

Reply:
The idea here is to account for non-tenanted requests (e.g. createTenant, listTenants, other pagination requests). But this is moot since our call where the tenant id is to be expected in any request.